march_rqt_gait_generator/src/march_rqt_gait_generator/gait_generator_view.py

```diff
@@ -0,0 +1,231 @@
+        loadUi(ui_file, self._widget)
+
+        context.add_widget(self._widget)
+
+        self.rviz_frame = self.create_rviz_frame()
+        self._widget.RvizFrame.layout().addWidget(self.rviz_frame, 1, 0, 1, 3)
+
+        # Store ui elements.
+        self.change_gait_directory_button = self._widget.SettingsFrame.findChild(QPushButton, 'ChangeGaitDirectory')
+        self.import_gait_button = self._widget.SettingsFrame.findChild(QPushButton, 'Import')
+        self.export_gait_button = self._widget.SettingsFrame.findChild(QPushButton, 'Export')
+        self.publish_gait_button = self._widget.SettingsFrame.findChild(QPushButton, 'Publish')
+        self.start_button = self._widget.RvizFrame.findChild(QPushButton, 'Start')
+        self.stop_button = self._widget.RvizFrame.findChild(QPushButton, 'Stop')
+        self.invert_button = self._widget.RvizFrame.findChild(QPushButton, 'Invert')
+        self.undo_button = self._widget.RvizFrame.findChild(QPushButton, 'Undo')
+        self.redo_button = self._widget.RvizFrame.findChild(QPushButton, 'Redo')
+        self.playback_speed_spin_box = self._widget.RvizFrame.findChild(QSpinBox, 'PlaybackSpeed')
+        self.height_left_line_edit = self._widget.RvizFrame.findChild(QLineEdit, 'HeightLeft')
+        self.height_right_line_edit = self._widget.RvizFrame.findChild(QLineEdit, 'HeightRight')
+        self.heel_distance_line_edit = self._widget.RvizFrame.findChild(QLineEdit, 'HeelHeelDistance')
+        self.topic_name_line_edit = self._widget.SettingsFrame.findChild(QLineEdit, 'TopicName')
+        self.gait_name_line_edit = self._widget.GaitPropertiesFrame.findChild(QLineEdit, 'Gait')
+        self.subgait_name_line_edit = self._widget.GaitPropertiesFrame.findChild(QLineEdit, 'Subgait')
+        self.version_name_line_edit = self._widget.GaitPropertiesFrame.findChild(QLineEdit, 'Version')
+        self.description_line_edit = self._widget.GaitPropertiesFrame.findChild(QLineEdit, 'Description')
+        self.gait_type_combo_box = self._widget.GaitPropertiesFrame.findChild(QComboBox, 'GaitType')
+        self.gait_type_combo_box.addItems(['walk_like', 'sit_like', 'stairs_like'])
+        self.duration_spin_box = self._widget.GaitPropertiesFrame.findChild(QDoubleSpinBox, 'Duration')
+        self.mirror_check_box = self._widget.SettingsFrame.findChild(QCheckBox, 'Mirror')
+        self.mirror_key1_line_edit = self._widget.SettingsFrame.findChild(QLineEdit, 'Key1')
+        self.mirror_key2_line_edit = self._widget.SettingsFrame.findChild(QLineEdit, 'Key2')
+        self.velocity_plot_check_box = self._widget.SettingsFrame.findChild(QCheckBox, 'ShowVelocityPlot')
+        self.effort_plot_check_box = self._widget.SettingsFrame.findChild(QCheckBox, 'ShowEffortPlot')
+        self.time_slider = self._widget.RvizFrame.findChild(QSlider, 'TimeSlider')
+        self.scale_setpoints_check_box = self._widget.GaitPropertiesFrame.findChild(QCheckBox, 'ScaleSetpoints')
```

Comment:
Is this really necessary? In other rqt widgets, we can access the objects like so:
```python
self._widget.time_slider.connect(...)
```
Without having to `findChild`. See [`march_rqt_note_taker.notes_widget.py`](https://github.com/project-march/monitor/blob/1627b9a99b7119bbcc117590ebe73c7ed678d920/march_rqt_note_taker/src/march_rqt_note_taker/notes_widget.py#L25-L39)

Reply:
Nice! This made things better :)